Eatery Blue/UI/MenusScreen/MenusViewController.swift

```diff
@@ -152,42 +153,39 @@ class MenusViewController: UIViewController {
 
     /// Updates the provided day's eateries from networking first, and loads the others in the background
     private func updateAllEateriesFromNetworking(withPriority day: Int = 0) async {
+        // Start the high-priority day first
+        async let highPriorityTask: Void = {
+            do {
+                try await updateAllEateriesByDayFromNetworking(day)
+            } catch {
+                logger.error("\(#function): \(error)")
+            }
+        }()
 
-        // Start Loading low priority days
-        Task.detached(priority: .low) { [weak self] in
-            guard let self else { return }
-
-            await withTaskGroup(of: Void.self) { [weak self] group in
+        // Start the low-priority days after
+        for i in 0...(daysToShow - 1) {
+            if i == day { continue }
+            Task(priority: .low) { [weak self] in
                 guard let self else { return }
 
-                for i in 0...(daysToShow - 1) {
-                    i != day ? group.addTask { [weak self] in
-                        guard let self else { return }
-
-                        do {
-                            try await updateAllEateriesByDayFromNetworking(i)
-                            let loading = await isLoading
-                            if await i == selectedIndex && loading { await stopLoading() }
-                        } catch {
-                            logger.error("\(#function): \(error)")
-                        }
-                    } : nil
+                do {
+                    try await updateAllEateriesByDayFromNetworking(i)
+                    if i == selectedIndex && isLoading { stopLoading() }
+                } catch {
+                    logger.error("\(#function): \(error)")
                 }
             }
         }
 
-        // Load the high priority day
-        do {
-            try await updateAllEateriesByDayFromNetworking(day)
-        } catch {
-            logger.error("\(#function): \(error)")
-        }
+        // Ensure the high-priority task completes before allowing function to return
+        await highPriorityTask
     }
```

Comment:
I see the vision here and I love you using async let. I learned about that fairly recently but I think you could achieve this in a more readable and cleaner way. If you want try something like this out and see if does the same thing.
```suggestion
private func updateAllEateriesFromNetworking(withPriority priorityDay: Int = 0) async {
    let tasks = (0..<daysToShow).map { day -> Task<Void, Error> in
        let priority: TaskPriority = day == priorityDay ? .userInitiated : .utility
        return Task(priority: priority) { [weak self] in
            guard let self else { return }
            try await updateAllEateriesByDayFromNetworking(day)
            
            if day == selectedIndex && isLoading {
                await MainActor.run { stopLoading() }
            }
        }
    }
    
    // Wait for all tasks to complete
    for task in tasks {
        try? await task.value
    }
}
```